### PR TITLE
[Cmake] Remove ChemistryLib and iphreeqc make for HeatTransportBHE process

### PR DESCRIPTION
**Pull request by ChaofanChen**
As titled.

### CODE REVIEW

**Comment by endJunction** [comment]
What's the reason?

**Comment by ChaofanChen** [comment]
> What's the reason? Could you add an explanation to the commit message.
> What has been changed is visible by the changes themselves.

Sorry for that, what I am understanding is that before what I changed in this PR, if I want to build the HeatTransportBHE process, the ChemistryLib and iphreeqc will be compiled together and linked, but this BHE process is not related to the chemical thing. So I remove it from corresponding cmake lists. If my understanding is wrong, please let me know. Also, the explanation is added in the commit message.

File: ProcessLib/CMakeLists.txt [hunk 1]
```diff
@@ -31,8 +31,7 @@ target_link_libraries(ProcessLib
 
 if (OGS_BUILD_PROCESS_ComponentTransport
     OR OGS_BUILD_PROCESS_RichardsComponentTransport
```
Comment (TomFischer) [inline ProcessLib/CMakeLists.txt]: Why not remove one of the double lines `OR OGS_BUILD_PROCESS_RichardsComponentTransport`?

File: ThirdParty/CMakeLists.txt [hunk 2]
```diff
@@ -11,8 +11,7 @@ option(OGS_USE_OPTIONAL_SUBMODULES "Option for enabling optional submodules" OFF
 # iphreeqc
 if (OGS_BUILD_PROCESS_ComponentTransport
     OR OGS_BUILD_PROCESS_RichardsComponentTransport
```
Comment (TomFischer) [inline ThirdParty/CMakeLists.txt]: Why not
```suggestion
```

File: ProcessLib/CMakeLists.txt (hunk 1 above)
Comment (ChaofanChen) [reply ProcessLib/CMakeLists.txt]: Removed.

File: ThirdParty/CMakeLists.txt (hunk 2 above)
Comment (ChaofanChen) [reply ThirdParty/CMakeLists.txt]: Removed.

**Comment by endJunction** [comment]
@ChaofanChen Thanks for the explanation.

**Pull request merged**